numpy/lib/npyio.py

```diff
@@ -733,10 +733,15 @@ def _getconv(dtype):
     """ Find the correct dtype converter. Adapted from matplotlib """
 
     def floatconv(x):
-        x.lower()
-        if '0x' in x:
-            return float.fromhex(x)
-        return float(x)
+        try:
+            return float(x)  # The fastest path.
+        except ValueError as exc:
+            try:
+                if '0x' in x:  # Don't accidentally convert "a" ("0xa") to 10.
+                    return float.fromhex(x)
+            except ValueError:
+                pass
+            raise exc  # Raise the original exception, which makes more sense.
```

Comment:
```suggestion
            raise  # Raise the original exception, which makes more sense.
```

Reply:
It wasn't clear to me that the original exception would indeed be reraised (not the new one), but I checked that that's the case :)